common/src/Utilities/PString.cpp

```diff
@@ -0,0 +1,80 @@
+#include "PrecompiledHeader.h"
+#include "PString.h"
+#include <codecvt>
+#include <wchar.h>
+
+#ifdef __cpp_lib_char8_t
+PString::operator std::u8string()
+{
+	std::u8string str(string.begin(), string.end());
+	return str;
+}
+#endif
+
+
+#ifdef _WIN32
+// Non default
+PString::PString(const std::wstring& utf16_string)
+{
+	const int size = WideCharToMultiByte(CP_UTF8, 0, utf16_string.c_str(), utf16_string.size(), nullptr, 0, nullptr, nullptr);
+	std::string converted_string(size, 0);
+	WideCharToMultiByte(CP_UTF8, 0, utf16_string.c_str(), utf16_string.size(), converted_string.data(), converted_string.size(), nullptr, nullptr);
+	string = converted_string;
+}
+
+PString::operator std::wstring()
+{
+	const int size = MultiByteToWideChar(CP_UTF8, 0, string.c_str(), string.size(), nullptr, 0);
+	std::vector<wchar_t> converted_string(size);
+	MultiByteToWideChar(CP_UTF8, 0, string.c_str(), string.size(), converted_string.data(), converted_string.size());
+	return converted_string.data();
```

Comment:
Any reason this isn't converted to `std::wstring` directly like the constructor above uses `std::string`?